object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -765,13 +761,19 @@ private void propagate(
   private void checkMustCall(
       LocalVarWithAssignTree assign, CFStore store, String outOfScopeReason) {
     List<String> mustCallValue = getMustCallValue(assign.assignTree);
+    // optimization: if there are no must-call methods, we do not need to perform the check
+    if (mustCallValue.isEmpty()) {
+      return;
+    }
     AnnotationMirror dummyCMAnno = createCalledMethods(mustCallValue.toArray(new String[0]));
 
     boolean report = true;
 
     AnnotationMirror cmAnno;
 
-    CFValue lhsCFValue = store.getValue(assign.localVar);
+    // sometimes the store is null!  this looks like a bug in checker dataflow.
```

Comment:
(Starting a new thread about this same line of code.)
I don't think this is a bug in the Checker Framework.  In your test case, the null store comes from calling getStoreAfter("return n").  There is no control flow after the return statement.  So `mustCallTraverse` could avoid calling getStoreAfter unless there are dataflow successor blocks, or it can handle the possibility that the store is null.

Reply:
It is possible this is not a bug but that we have misunderstood the spec of the `getStoreAfter` API.  However, the `return n` node in the CFG does have a successor; the relevant code calling `getStoreAfter` runs inside a conditional that checks if the successor is an exit node:

https://github.com/msridhar/object-construction-checker/blob/9d2075db86bba301179fb5585d8204babda98bb8/object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java#L532-L549

This same code calls `getStoreAfter` for pretty much any `return` statement without getting back a null store.  

Also, in `MustCallNullStore.java`, if I comment out lines 15, 17, and 18 to remove the `try-finally` construct, I no longer get an NPE, which is suspicious.  If I had time to dig deeper I would try to look at how the control-flow graph looks for the `inflateDirect` method to see if there is anything suspicious there.